wgpu-core/src/command/render.rs

```diff
@@ -110,6 +110,15 @@ pub enum RenderCommand {
         depth_max: f32,
     },
     SetScissor(Rect<u32>),
+    SetPushConstant {
+        stages: wgt::ShaderStage,
+        offset: u32,
+        size_bytes: u32,
+        /// None means there is no data and the data should be an array of zeros.
+        ///
+        /// Facilitates clears in renderbundles which explicitly do their clears.
+        values_offset: Option<usize>,
```

Comment:
does this need to be `usize`? I.e. can we say that you aren't going to have more than 64k of push constants ever, and have this offset being `u16` instead? We'd want to keep the `enum RenderCommand` size to be small. Even though right now it may not be the enum variant that makes it big, we don't want to face this later down the road when we try to shrink it.

Reply:
I'm going to make this a u32, as this index can be as large as the sum of all push constants used in a render pass, and if someone sets a lot of push constants (or used metal's 4k to their full advantage), it would overrun a u16 pretty quickly.